ksql-engine/src/main/java/io/confluent/ksql/util/HybridQueryIdGenerator.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.util;
+
+
+public class HybridQueryIdGenerator implements QueryIdGenerator {
```

Comment:
Would benefit from a javadoc explaining why we need this hybrid approach, i.e. because we changed the way we generate ids and we need to support both.

Also, can you raise a github issue against the next breaking change release, i.e. by setting 6.0 as the milestone, to remove the need for this hybrid class.  Please reference this PR from the issue, so people can get context.

Reply:
https://github.com/confluentinc/ksql/issues/3392